pkg/services/provisioning/values/values.go

```diff
@@ -0,0 +1,178 @@
+// A set of value types to use in provisioning. They add custom unmarshaling logic that puts the string values
+// through os.ExpandEnv.
+package values
+
+import (
+	"github.com/pkg/errors"
+	"os"
+	"reflect"
+	"strconv"
+)
+
+// Can be changed for testing
+var getEnv = os.Getenv
```

Comment:
Instead of mocking `os.Getenv` we could set env variables in the tests.
https://gobyexample.com/environment-variables

I think that's cleaner.

Reply:
Yes that looks better will change.